leader-election-impl/src/test/java/com/palantir/leader/LeadershipElectionCheckTest.java

```diff
@@ -71,57 +71,69 @@ public void setup() {
+        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthReport().status())
                 .isEqualTo(LeaderElectionHealthStatus.HEALTHY);
+        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthReport().leaderElectionRate())
+                .isGreaterThan(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
+    }
+
+    @Test
+    public void shouldBeHealthyForOneLeaderElectionPerMinute() {
+        markLeaderElectionsAtSpecifiedInterval(CLIENT_1, 20, Duration.ofSeconds(60), true);
+        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthReport().leaderElectionRate())
+                .isLessThanOrEqualTo(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
     }
 
     @Test
     public void shouldBeUnhealthyForMoreThanOneLeaderElectionPerMinute() {
-        markLeaderElectionsAtSpecifiedInterval(CLIENT_1, 5,  Duration.ofSeconds(10));
-        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthStatus())
-                .isEqualTo(LeaderElectionHealthStatus.UNHEALTHY);
+        markLeaderElectionsAtSpecifiedInterval(CLIENT_1, 5,  Duration.ofSeconds(10), true);
+        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthReport().leaderElectionRate())
+                .isGreaterThan(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
     }
 
     @Test
     public void shouldBeHealthyForOneLeaderElectionPerMinuteAcrossClients() {
-        markLeaderElectionsAtSpecifiedInterval(CLIENT_1, 5,  Duration.ofSeconds(60));
-        markLeaderElectionsAtSpecifiedInterval(CLIENT_2, 5,  Duration.ofSeconds(60));
+        markLeaderElectionsAtSpecifiedInterval(CLIENT_1, 5,  Duration.ofSeconds(60), true);
+        markLeaderElectionsAtSpecifiedInterval(CLIENT_2, 5,  Duration.ofSeconds(60), false);
 
-        assertThat(leaderElectionHealthCheckForOnlyClient1.leaderElectionRateHealthStatus())
-                .isEqualTo(LeaderElectionHealthStatus.HEALTHY);
+        assertThat(leaderElectionHealthCheckForOnlyClient1.leaderElectionRateHealthReport().leaderElectionRate())
+                .isLessThanOrEqualTo(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
 
-        assertThat(leaderElectionHealthCheckForOnlyClient2.leaderElectionRateHealthStatus())
-                .isEqualTo(LeaderElectionHealthStatus.HEALTHY);
+        assertThat(leaderElectionHealthCheckForOnlyClient2.leaderElectionRateHealthReport().leaderElectionRate())
+                .isLessThanOrEqualTo(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
 
-        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthStatus())
-                .isEqualTo(LeaderElectionHealthStatus.HEALTHY);
+        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthReport().leaderElectionRate())
+                .isLessThanOrEqualTo(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
     }
 
     @Test
     public void shouldBeUnhealthyOverallEvenIfIndividualClientsAreHealthy() {
```

Comment:
This test makes sense for now. The aggregation and/or threshold probably needs revisiting if timelock partitioning is introduced.

Reply:
💯